get-indirection-metrics/index.js

```diff
@@ -0,0 +1,107 @@
+const { Octokit } = require("@octokit/rest");
+const fs = require("fs");
+const dig = require("node-dig-dns");
+
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+});
+
+const INDIRECTION_REPO = "isomerpages/isomer-indirection";
+
+async function getAllIndirectionDomains() {
+  const dnsTree = await octokit.git.getTree({
+    owner: INDIRECTION_REPO.split("/")[0],
+    repo: INDIRECTION_REPO.split("/")[1],
+    tree_sha: "main",
+    recursive: 1,
+  });
+
+  const dnsFiles = dnsTree.data.tree.filter((file) =>
+    file.path.startsWith("dns/")
+  );
+
+  const domains = dnsFiles.map((file) => file.path.split("/")[1].slice(0, -3));
+
+  return domains;
+}
+
+async function getDomainCNAMERecord(domain) {
+  let result = await dig([domain, "CNAME"]);
+  if (!Object.keys(result).includes("answer") || result.answer.length === 0) {
+    result = await dig([`www.${domain}`, "CNAME"]);
+  }
+
+  return result;
+}
+
+async function getDomainsOnIndirectionLayer(domains) {
+  const domainsOnIndirectionLayer = [];
+  process.stdout.write(`Checking domain 0/${domains.length}...`);
+
+  for (let i = 0; i < domains.length; i++) {
+    const result = await getDomainCNAMERecord(domains[i]);
+    process.stdout.clearLine(0);
+    process.stdout.cursorTo(0);
+    process.stdout.write(`Checking domain ${i + 1}/${domains.length}...`);
+
+    if (
+      Object.keys(result).includes("answer") &&
+      result.answer.length > 0 &&
+      result.answer.some((record) =>
+        record.value.endsWith(".hostedon.isomer.gov.sg.")
+      )
+    ) {
+      domainsOnIndirectionLayer.push(domains[i]);
+    }
+  }
+
+  return domainsOnIndirectionLayer;
+}
+
+(async function () {
+  try {
+    const domains = await getAllIndirectionDomains();
+    const domainsOnIndirectionLayer = await getDomainsOnIndirectionLayer(
+      domains
+    );
+    const domainsNotOnIndirectionLayer = domains.filter(
+      (domain) => !domainsOnIndirectionLayer.includes(domain)
+    );
+
+    fs.writeFileSync("indirection-domains.log", domains.join("\n"));
+    fs.writeFileSync(
+      "domains-on-indirection.log",
+      domainsOnIndirectionLayer.join("\n")
+    );
```

Comment:
the naming here is confusing (`indirection-domains` and `domains-on-indirection`), maybe we might wanna use `hosted-on-isomer` or add a comment for clarity's sake

Reply:
Updated in ba907ffbc01dcb04a72ead1e268372e2e31720cf. The details are captured inside the README.md file.